build/scripts/obtain_dotnet.sh

```diff
@@ -27,7 +15,16 @@ install_dotnet () {
     local DOTNET_VERSION="$(get_tool_version dotnetSdk)"
     local DOTNET_PATH="${THIS_DIR}"/../../Binaries/dotnet-cli
 
-    if [[ ! -x "${DOTNET_PATH}/dotnet" ]]
+    # check if the correct `dotnet` is already on the PATH
+    if command -v dotnet >/dev/null 2>&1
+    then
+        if [[ "${FORCE_DOWNLOAD:-false}" != true && "$(dotnet --version)" = "${DOTNET_VERSION}" ]]
+        then
+            return 0
+        fi
+    fi
+
+    if [[ ! -x "${DOTNET_PATH}/dotnet" || "$(${DOTNET_PATH}/dotnet --version)" != "${DOTNET_VERSION}" ]]
```

Comment:
What does `! -x` do?

Reply:
`-x file`: "file" exists and is executable (the permission bit)

`! expr`: expr is false

`! -x file`: file does not exist, or is not executable